libsupport/src/ArrowInterchange.cpp

```diff
@@ -70,3 +109,39 @@ katana::EmptyChunkedArray(
   std::vector<std::shared_ptr<arrow::Array>> chunks{maybe_array.ValueOrDie()};
   return std::make_shared<arrow::ChunkedArray>(chunks);
 }
+
+void
+katana::DiffFormatTo(
+    fmt::memory_buffer* buf, const std::shared_ptr<arrow::ChunkedArray>& a0,
+    const std::shared_ptr<arrow::ChunkedArray>& a1,
+    size_t approx_total_characters) {
+  if (a0->type() != a1->type()) {
+    fmt::format_to(
+        *buf, "Arrays are different types {}/{}\n", a0->type()->name(),
+        a1->type()->name());
+    return;
+  }
+  auto b0 = Unchunk(a0).value();
+  auto b1 = Unchunk(a1).value();
```

Comment:
Preference for checking these results

Reply:
Thank you for catching this!  Darn it, I just decided to try a "high productivity" prototyping mode where I aggressively did not check errors.  Yeah, maybe that way of working is appropriate for some, but I need to stick to my slower but more careful ways.